### PR TITLE
Improve stability

### PR DESCRIPTION
**Pull request by jakelauer**
This PR fixes a couple of issues I ran into while using this tool.

- Enables partial uploads so that you can determine if there is a problem early, rather than waiting for the entire scan to finish
- Fixes a place that was failing to pass the playlist_id
- Other minor fixes

### CODE REVIEW

**Comment by BoscoDomingo** [comment]
Thank you so very much for this!! We love that our small app helped you and I'll take a look at your code and make sure everything works as intended as soon as I get some time this week. I ought to set up GitHub Actions or something, might get around to that one day , but for now manual review it is ;)

**Review by BoscoDomingo — CHANGES_REQUESTED** [review]
So the following error appeared at the very end. Should be a relatively easy fix (variable not declared)
image

File: main.py [hunk 1]
```diff
@@ -140,33 +147,71 @@ def add_tracks_to_playlist(track_ids):
...
-            else:
-                track_ids.append(result)
+    if os.path.exists(cached_playlist_id_filename):
+        playlist_file = open(cached_playlist_id_filename)
+        playlist_id = playlist_file.read()
+    else:
+        playlist_id = ensure_playlist_exists(playlist_id)
+        playlist_file = open(cached_playlist_id_filename, "w+")
+        playlist_file.write(playlist_id)
+        playlist_file.close()
```
Comment (BoscoDomingo) [inline main.py:164]: Could you move this to a function? Something like `cache_playlist_id()` or similar

File: main.py [hunk 2]
```diff
@@ -140,33 +147,71 @@ def add_tracks_to_playlist(track_ids):
...
+
+    if os.path.exists(successful_matches_filename):
+        append_write = 'rb+' # append if already exists
+    else:
+        append_write = 'wb+'
```
Comment (BoscoDomingo) [inline main.py:169]: Same here, make it a function like `decide_append_write_mode()`

**Review by BoscoDomingo — CHANGES_REQUESTED** [review]
~I'll be leaving a PR in your fork with some changes~ Nevermind, I don't have push access to your repo, so I created a new branch in this one, `PRs/#4`, although there are still some tweaks I'd like you to do regardless to approve this one and merge it. Thanks again for your time and contribution :)

**Comment by BoscoDomingo** [comment]
image
This is with my changes in the aforementioned `PRs/#4` where you can see I had some songs already cached + some new ones, and the numbers are weird, so I definitely didn't fix it 100%.

**Comment by jakelauer** [comment]
I'll take a look at this when I get a chance!

**Comment by jakelauer** [comment]
PS, thank you for writing this, it's literally the only thing on the internet I could find that would let me move my music to Spotify the way I wanted.

**Comment by jakelauer** [comment]
PPS I am a JavaScript dev by trade and literally have never written a line of python in my life before this PR, so keep that in mind! I'm probably not the best person to make something pretty and nice. But I can give it a whack.

**Comment by BoscoDomingo** [comment]
> 
> 
> PS, thank you for writing this, it's literally the only thing on the internet I could find that would let me move my music to Spotify the way I wanted.

Aboslutely, I'm glad it could help someone else other than me, it's the main reason why I put it out there!!

Also I'm quite impressed you jumped onto a different language like that, massive props to you for figuring it out and making it work. I totally understand it not being the absolute most pristine thing if it's your first ever go at Python lol. Most people (including me in this specific case) don't even get it right after becoming "experts" 🤣 Don't fret about it. The fact you contributed is already way above and beyond what most people even bother doing

**Pull request closed without merging**